doc/dev/binary-file-formats.rst

```diff
@@ -0,0 +1,94 @@
+any external compression/encoding (e.g. gzip encapsulation) is removed. Note
+that this makes them incompatible with older versions of Khmer.
+
+Countgraph
+----------
+
+(a.k.a ``CountingHash``, a Count-min Sketch)
+
+:Preferred extension: '.ct' (count table)
+
+The header is in the format below, again in the order of file offset. C++
+macro definitions are given in parenthesis. ``Len`` is the field's size, in
+bytes, and ``Off`` is the field's zero-based byte offset in the file.
+
+================== ===== ===== ==============================================
+Field               Len   Off     Value
+================== ===== ===== ==============================================
+Magic string        4       0   ``OXLI`` (``SAVED_SIGNATURE``)
+Version             1       4   ``0x04`` (``SAVED_FORMAT_VERSION``)
+File Type           1       5   ``0x01`` (``SAVED_COUNTING_HT``)
+Use Bigcount        1       6   ``0x01`` if bigcounts is used, else ``0x00``
+K-size              1       7   k-mer length, ``1 <= k <= 32``
+Number of Tables    1       8   Number of Count-min Sketch tables
+================== ===== ===== ==============================================
+
+Then follows the Countgraph's tables. For each table:
+
+================== ===== ===== ==============================================
+Field               Len   Off     Value
+================== ===== ===== ==============================================
+Table size          8       0   Length of this table. (``uint64_t``)
+Bins                N       8   This table's bins, length given by previous field. (``uint8_t``)
+================== ===== ===== ==============================================
+
+Nodegraph
+---------
+
+(a.k.a ``HashBits``, a Bloom Filter)
+
+:Preferred extension: '.pt' (presence table)
+
+The header is in the format below, again in the order of file offset. Value
+macro definitions are given in parenthesis
+
+================== ===== ===== ==============================================
+Field               Len   Off     Value
+================== ===== ===== ==============================================
+Magic string        4       0   ``OXLI`` (``SAVED_SIGNATURE``)
+Version             1       4   ``0x04`` (``SAVED_FORMAT_VERSION``)
+File Type           1       5   ``0x02`` (``SAVED_HASHBITS``)
+K-size              4       6   k-mer length, as ``unsigned int``.
+Number of Tables    1      10   Number of Nodegraph tables.
+================== ===== ===== ==============================================
+
+Then follows the Nodegraph's tables. For each table:
+
+================== ======= ===== ==============================================
+Field               Len     Off     Value
+================== ======= ===== ==============================================
+Table size          8         0   Length of table, **in bits**. (``uint64_t``)
+Bins                N/8+1     8   This table's bytes, length given by previous field, divided by 8, plus 1. (``uint8_t``)
```

Comment:
@mr-c Is there a way of wrapping lines in RST simple tables that doesn't create a new row? Or do I have to suck it up and make a proper table?